python/openmlconverter/src/converter.py

```diff
@@ -0,0 +1,151 @@
+        "dateModified": _ds(field="processing_date", transform=dateutil.parser.parse),
+        "datePublished": _ds(field="collection_date", transform=_lenient_date_parser),
+        "inLanguage": _ds(field="language", transform=lambda v: langcodes.find(v).language),
+        "isAccessibleForFree": True,
+        "license": _ds(field="license"),
+        "creativeWorkStatus": _ds(field="status"),
+        "keywords": _ds(field="tag"),
+        "citation": _ds(field="citation"),
+        "sameAs": _ds(field="original_data_url"),
+        "url": f"https://www.openml.org/api/v1/json/data/{_ds(field='id')}",
+        "distribution": distributions,
+        "recordSet": [
+            {
+                "name": _ds(field="name"),
+                "@type": "ml:RecordSet",
+                "source": f"#{{{distribution_source}}}",
+                "key": _row_identifier(openml_features),
+                "field": [
+                    {
+                        "name": feat["name"],
+                        "@type": "ml:Field",
+                        "dataType": _datatype(feat["data_type"], feat.get("nominal_value", None)),
+                        "source": f"#{{{distribution_source}/{feat['name']}}}",
+                    }
+                    for feat in openml_features
+                ],
+            }
+        ],
+    }
+    _remove_none_values(croissant)
+    for recordSet in croissant["recordSet"]:
+        _remove_none_values(recordSet)
+    return croissant
+
+
+def _get_field(json_dict, field: str, transform=None, required=False):
+    """
+    Convenience function. Get field and optionally perform the transformation. If field does not
+    exist, throw an error if the field is  required, otherwise return None.
+    """
+    if field not in json_dict:
+        if required:
+            raise ValueError(f"Required field {field} missing.")
+        return None
+    val = json_dict[field]
+    if transform is not None:
+        return transform(val)
+    return val
+
+
+def _person(name: str) -> dict | None:
+    if name is None or name == "":
+        return None
+    return {"@context": "https://schema.org", "@type": "Person", "name": name}
+
+
+def _distribution(name: str, url: str) -> dict | None:
+    if url is None:
+        return None
+
+    if url.endswith(".arff"):
+        type_ = "arff"
+        mimetype = "text/plain"  # No official arff mimetype exists
+    elif url.endswith(".pq"):
+        type_ = "parquet"
+        mimetype = "application/vnd.apache.parquet"  # Not an official mimetype yet
+        # see https://issues.apache.org/jira/browse/PARQUET-1889
+    else:
+        raise ValueError(f"Unrecognized file extension in url: {url}")
+    return {
+        "name": f"{name} ({type_})",
+        "@type": "FileObject",
+        "contentUrl": url,
+        "encodingFormat": mimetype
+        # TODO: add the md5 hash? But where? 'sha256' is in the Google Docs file, but not official?
```

Comment:
https://schema.org/sha256 is pending implementation feedback but I suppose we can use it.
TFDS also uses MD5, and while re-computing the sha256 is possible, I don't know how much friction it would add to the various actors...

Reply:
I resolve this conversation for now, moved it to a separate issue: https://github.com/mlcommons/datasets_format/issues/8